src/Microsoft.Azure.WebJobs.Host/Listeners/FunctionListener.cs

```diff
@@ -47,12 +48,23 @@ public void Dispose()
             catch (Exception e)
             {
                 new FunctionListenerException(_descriptor.ShortName, e).TryRecover(_trace);
+
+                // if recovered, disable and attempt a best effort stop of the listener
+                _disabled = true;
+                try
+                {
+                    await _listener.StopAsync(cancellationToken);
```

Comment:
Not sure we should call Stop here in Start. Either we just don't call Stop at all if we never Started completely (which seems correct to me), or you move this Stop handling into the Stop method below.